src/modules/comment/comment.controller.ts

```diff
@@ -0,0 +1,101 @@
+import { Router } from 'express'
+import { CommentService } from './comment.service'
+import {
+  CommentOwnerMiddleware,
+  IsUUID,
+  TokenParserMiddleware,
+  validateReqBody,
+} from '../../shared'
+import { createCommentSchema } from './schemas'
+import { CommentOwnerOrAdminMiddleware } from '../../shared/middlewares/CommentOwnerOrAdmin.middleware'
+
+export const commentRouter = Router()
+const commentService = new CommentService()
+
+commentRouter.post(
+  '/blogs/:blog_id/comments',
+  validateReqBody(createCommentSchema),
+  TokenParserMiddleware,
+  async (req: any, res) => {
+    try {
+      const { content } = req.body
+      const { id: author_id } = req.user
+      const { blog_id } = req.params
+
+      if (!blog_id || !author_id || !content || !content.trim()) {
+        return res.status(400).json({ message: 'Error with some fields' })
+      } else {
+        const comment = await commentService.createComment({
+          author_id,
+          blog_id,
+          content: content.trim(),
+        })
+        res.status(201).json(comment)
+      }
+    } catch (error: any) {
+      res.status(400).json({ message: error.message })
+    }
+  },
+)
+
+commentRouter.get('/blogs/:blog_id/comments', async (req, res) => {
```

Comment:
Suggestion: You can add pagination to comments too. in UI it is gonna be Load more comments and it will work via pagination

Reply:
I did a pagination